Lesson16/Lesson16-Busra/busra.js

```diff
@@ -0,0 +1,442 @@
+
+
+
+/*
+15. Filter Words Longer Than X
+    - Define a function `filterLongWords(words, minLength)` that uses a for loop
+      to collect only the words that have a length >= minLength.
+    - Log the resulting array.
+*/
+const myWords=["bir","iki","üç","dört","beş","altı","yedi","sekiz"];
+
+
+
+function filterLongWords(words, minLength){
+  let newWords = [];
+  for(let i=0; i < words.length; i++){
+    if(words[i].length >= minLength){
+      newWords.push(words[i]);
+    }
+    
+  }
+  return newWords;
+}
+
+filterLongWords(myWords, 3);
+
+
+/*
+16. Log Array Elements with Their Indices
+    - Define a function `logElementsWithIndex(arr)` that loops through the array
+      and logs "Index: i, Value: arr[i]" for each element.
+*/
+
+const myArrayEx16 = ["busra","demirkesen",34,true,false];
+function logElementsWithIndex(arr){
+  for(let i=0; i<myArrayEx16.length; i++){
+    console.log(i,myArrayEx16[i]);
+  }
+
+}
+
+logElementsWithIndex(myArrayEx16);
+
+
+/*
+17. Find the Smallest Number in an Array
+    - Define a function `findMin(numbers)` that loops through the array
+      to find and return the smallest number.
+    - Log the smallest number.
+*/
+const MyNumbersEx17 = [4,9,2,1,7,15,3,11];
+
+
+function findMin(numbers){
+  let smallestNumber = numbers[0];
+  for(let i=0; i < numbers.length; i++){
+    if(numbers[i] < smallestNumber){
+      smallestNumber= numbers[i];
+    }
+  }
+  return smallestNumber;
+}
+findMin(MyNumbersEx17);
+
+
+
+/*
+18. Count Occurrences of a Word in an Array
+    - Define a function `countOccurrences(arr, word)` that loops through `arr`
+      to count how many times `word` appears.
+    - Log the count.
+
+*/
+
+const foods=["apple","egg","banana","apple","bread","apple"];
+
+
+function countOccurrences(arr, word){
+  let counterEx18 = 0;
+    for(let i =0; i < arr.length; i++){
+        if(arr[i].toLowerCase() === word.toLowerCase()){
+          counterEx18 +=1;
+        }
+    }
+    return counterEx18;
+}
+
+
+countOccurrences(foods,"apple");
+
+
+/*
+19. Remove Falsy Values
+    - Define a function `removeFalsyValues(arr)` that loops through an array
+      and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+    - Log the new array.
+*/
+const myArrayEx19=["string", 3 , "number", 8, false, 0 ,"", null, undefined, NaN];
+
+function removeFalsyValues(arr){
+   let myNewArrayEx19=[];
+   for(let i=0; i < arr.length; i++){
+    if(arr[i]){
+      myNewArrayEx19.push(arr[i]);
+    }
+    return myNewArrayEx19;
+   }
+}
+removeFalsyValues(myArrayEx19);
+
+
+
+/*
+20. Sum of All Digits in a String
+    - Define a function `sumDigits(str)` that loops through each character of `str`,
+      checks if it's a digit, and if so, adds it to a total sum.
+    - Log the final sum.
+    - Example: "abc123" -> 6
+
+*/
+const strEx20 = "def432ne379";
+function sumDigits(str){
+  let countEx20 = 0;
+  for(let i=0; i<str.length; i++){
+    if(!isNaN(str[i]) && str[i] !== ' '){
```

Comment:
In this case, you only want to sum normal digits. Therefore, it's better to use this check:
```suggestion
    if(typeof str[i] === 'number'){
```